Codestyles

lparkermg opened the pull request.

lparkermg committed:
Updated projects with Xml docs and warnings as errors in release mode.

lparkermg committed:
Added Code Styles to projects.

lparkermg committed:
Implemented some of the code style issues in the core project.

lparkermg committed:
Implemented some of the code style issues in the Dispatchers.EventStore project.

lparkermg committed:
Implemented some of the code style issues in snapshotting.probuf.

lparkermg committed:
Added comments and docs to Snapshotting.Protobuf dispatcher.

lparkermg committed:
Updated documentation in MessageDispatcher core.

lparkermg committed:
Documented Dispatchers.EventStore project.

lparkermg committed:
Added a check into CatchupPercentage to prevent against DivideByZeroExceptions.

rcknight commented on src/Core/OnExceptionDispatchers/OnExceptionExitDispatcher.cs:
```diff
@@ -0,0 +1,27 @@
+    using System;
+
+    /// <summary>
+    /// Sets up a dispatcher which Exits when an exception is thrown.
```
=> something like "A wrapping message dispatcher which kills the application when an exception is thrown from the inner dispatcher."

rcknight commented on src/Core/OnExceptionDispatchers/OnExceptionExitDispatcher.cs:
```diff
@@ -0,0 +1,27 @@
+    /// <summary>
+    /// Sets up a dispatcher which Exits when an exception is thrown.
+    /// </summary>
+    /// <typeparam name="TMessage">Message Type</typeparam>
```
=> The type of message this dispatcher handles

rcknight commented on src/Core/OnExceptionDispatchers/OnExceptionExitDispatcher.cs:
```diff
@@ -0,0 +1,27 @@
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OnExceptionExitDispatcher{TMessage}"/> class.
+        /// </summary>
+        /// <param name="dispatcher">Dispatcher.</param>
```
These params need better descriptions

rcknight commented on src/Core/OnExceptionDispatchers/OnExceptionSkipOrExitDispatcher.cs:
```diff
@@ -0,0 +1,52 @@
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OnExceptionSkipOrExitDispatcher{TMessage}"/> class.
+        /// </summary>
+        /// <param name="dispatcher">Dispatcher to use.</param>
```
These need better descriptions. Something like the following maybe?

"The inner dispatcher which this dispatcher should wrap"
"A `Func` which decides what to do when an exception is thrown by the inner dispatcher. It should return `false` to instruct the dispatcher to skip the current message and continue processing future messages, and `true` if error is fatal and the application should exit.
"the onExitLog callback will be called prior to the application exiting in case of a fatal error (as indicated by `shouldExit`). This gives an opportunity to log the exception.

rcknight commented on src/Core/RetryingDispatchers/ExponentialRetryingDispatcher.cs:
```diff
@@ -0,0 +1,44 @@
+    using System;
+
+    /// <summary>
+    /// Sets up a dispatcher that exponentially retries.
```
Could be more detailed.

rcknight commented on src/Core/RetryingDispatchers/ExponentialRetryingDispatcher.cs:
```diff
@@ -0,0 +1,44 @@
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExponentialRetryingDispatcher{TMessage}"/> class.
+        /// </summary>
+        /// <param name="dispatcher">Dispatcher.</param>
```
Should provide more detail on lots of these.

rcknight replied:
I'm going to stop writing these everywhere, seems like all the docs could do with a review / rewording with more descriptive text.

rcknight commented on src/Core/RetryingDispatchers/RetryingDispatcher.cs:
```diff
@@ -1,31 +1,49 @@
         {
             _retryLimit = retryLimit;
-            _retryLogAction = retryLogAction ?? ((_, __) => { });
+            _retryLogAction = retryLogAction ?? ((_, e) => { });
```
Did codestyle make us change this?

rcknight commented on src/Dispatchers.EventStore/WriteThroughFileCheckpoint.cs:
```diff
@@ -0,0 +1,181 @@
+        private static class Filenative
+        {
+#pragma warning disable SA1310 // Field names should not contain underscore - Gets pulled from the eventstore DLLs
+            public const int FILE_FLAG_NO_BUFFERING = 0x20000000;
```
No reason we couldn't rename this.

rcknight commented on src/core/DeserializingMessageDispatcher.cs:
```diff
@@ -1,38 +1,84 @@
             var type = default(TLookupKey);
 
-            if (!TryGetMessageType(message, out type)) return;
+            if (!TryGetMessageType(message, out type))
```
could use out var?

rcknight commented on src/dispatchers.eventstore/EventStoreJObjectDispatcher.cs:
```diff
@@ -1,30 +1,50 @@
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Dispatcher that deserializes from a JObject.
```
TO a JObject

rcknight commented on src/dispatchers.eventstore/EventStoreSubscriber.cs:
```diff
@@ -160,18 +318,20 @@ public CatchupProgress CatchUpPercentage
 
-            //Make heartbeats optional!
-            //if (!heartbeatFrequency.HasValue)
+            // Make heartbeats optional!
```
Should delete these commented lines

rcknight reviewed (dismissed):
Couple code changes but the big issue for me is the xml docs.

IMO if we are going to bother writing docs we need to take a bit more care with them than this.

Commit:
General code cleanup

Commit:
Cleanup in EventStoreSubscriber

Commit:
Move rogue using declaration

Commit:
Improve EventStoreSubscriber stability.

In some weird situations, null reference exceptions were being thrown when restarting a subscriber, and preventing restarts; this has been fixed. The EventStoreSubscriber was also not trying to restart in situations where it should have, which has been fixed, and subscriptions were occasionally being duplicated; to prevent this, every time the Subscription is started and restarted, inside of the lock which contains the start logic, it is now ensured that no other subscription exists.

oscarhirst committed:
Updated the solution's documentation

oscarhirst committed:
Removed redundant code from WriteThroughFileCheckpoint.cs

oscarhirst committed:
Put back some of WriteThroughFileCheckpoint.cs' orginal code

rcknight committed:
Merge pull request #9 from cognisant/eventstore-subscriber-stability

Improve EventStoreSubscriber stability.
